CONTRIBUTING.rst

```diff
@@ -96,19 +116,21 @@ In order to update the version and change log entry, you have to create a
 release file. This is a normal restructured text file called RELEASE.rst that
 lives in the root of the repository and will be used as the change log entry.
 
-It should start with following lines:
-
-* RELEASE_TYPE: major
-* RELEASE_TYPE: minor
-* RELEASE_TYPE: patch
-
-This specifies the component of the version number that should be updated, with
-the meaning of each component following `semver <http://semver.org/>`_. As a
-rule of thumb if it's a bug fix it's probably a patch version update, if it's
-a new feature it's definitely a minor version update, and you probably
-shouldn't ever need to use a major version update unless you're part of the
-core team and we've discussed it a lot.
-
+The first line of the file specifies the component of the version number that
+will be updated, according to our `semantic versioning <http://semver.org/>`_
+policy.
+
+- ``RELEASE_TYPE: major`` is for breaking changes, and will only be used by the
+  core team after extensive discussion.
+- ``RELEASE_TYPE: minor`` is for anything that adds to the public (ie documented)
+  API, changes an argument signature, or adds a new deprecation or health check.
+  Minor (or patch) releases **may not** cause errors in any code that runs
```

Comment:
Given that this PR itself doesn't meet the letter of this rule (as you noted), maybe tweak it to allow this change.

Maybe something like this?:

> ...  **may not** cause errors in any code that runs correctly without errors on an earlier version (and even if the previous behavior was incorrect, we will still think carefully before causing new errors).

Reply:
Hmm. Interesting. I *thought* we had an exception listed somewhere for changing silent errors into noisy errors, but it seems we don't. We definitely should have that for adding new health checks if nothing else!